src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -44,56 +44,58 @@ public static class PackageDownloader
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            // While CancellationTokenSource has a dispose, if we call it before the tasks using
+            // the token are done we risk having these tasks fail with a object disposed exception so
+            // this should not be in a using unless we want to block until all tasks are done/cancled
+            var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```

Comment:
@jascam, But, this means that the linkedTokenSource does not get disposed. How do you solve that?

Note that disposing LinkedTokenSource will unlink the CTS from the tokens it was linked to. You should not get an ObjectDisposedException. I say this based on http://blogs.msdn.com/b/pfxteam/archive/2012/03/25/10287435.aspx (Look at comments from Stephen Toub).

When do you see that the task is throwing ObjectDisposedException? When one of the tasks succeed or when none of the tasks succeed.